framework/source/class/qx/log/appender/Formatter.js

```diff
@@ -0,0 +1,308 @@
+
+      var breakdown = this.normalizeEntryClass(entry);
+      var result = "";
+      if (breakdown.clazz) {
+        result += breakdown.clazz.classname;
+      }
+      if (breakdown.object) {
+        output.push("<span class='object' title='Object instance with hash code: " + 
+            breakdown.object.toHashCode() + "'>", breakdown.classname, "[", breakdown.object, "]</span>: ");
+      } else if (breakdown.hash) {
+        output.push("<span class='object' title='Object instance with hash code: " + 
+            breakdown.hash + "'>", breakdown.classname, "[", breakdown.hash, "]</span>: ");
+      } else if (breakdown.clazz) {
+        output.push("<span class='object'>" + breakdown.clazz.classname, "</span>: ");
+      }
+
+      var items = entry.items;
+      for (var i = 0, il = items.length; i < il; i++) {
+        item = items[i];
+        msg = item.text;
+
+        if (msg instanceof Array) {
+          var list = [];
+
+          for (var j = 0, jl = msg.length; j < jl; j++) {
+            sub = msg[j];
+            if (typeof sub === "string") {
+              list.push("<span>" + qx.log.appender.Formatter.escapeHTML(sub) + "</span>");
+            } else if (sub.key) {
+              list.push("<span class='type-key'>" + sub.key + "</span>:<span class='type-" + sub.type + "'>"
+                  + qx.log.appender.Formatter.escapeHTML(sub.text) + "</span>");
+            } else {
+              list.push("<span class='type-" + sub.type + "'>" + qx.log.appender.Formatter.escapeHTML(sub.text) + "</span>");
+            }
+          }
+
+          output.push("<span class='type-" + item.type + "'>");
+
+          if (item.type === "map") {
+            output.push("{", list.join(", "), "}");
+          } else {
+            output.push("[", list.join(", "), "]");
+          }
+
+          output.push("</span>");
+        } else {
+          output.push("<span class='type-" + item.type + "'>" + qx.log.appender.Formatter.escapeHTML(msg) + "</span> ");
+        }
+      }
+
+      var wrapper = document.createElement("DIV");
+      wrapper.innerHTML = output.join("");
+      wrapper.className = "level-" + entry.level;
+
+      return wrapper;
+    }
+  },
+  
+  
+  statics: {
+    /** @type {qx.util.format.DateFormat} format for datetimes */
+    __DATETIME_FORMAT: null,
+    
+    /** @type {qx.log.appender.Formatter} the default instance */
+    __defaultFormatter: null,
+    
+    /**
+     * Returns the default formatter
+     * 
+     * @return {qx.log.appender.Formatter}
+     */
+    getDefaultFormatter: function() {
+      if (!qx.log.appender.Formatter.__defaultFormatter) {
+        qx.log.appender.Formatter.__defaultFormatter = new qx.log.appender.Formatter();
+      }
+      return qx.log.appender.Formatter.__defaultFormatter;
+    },
+    
+    /**
+     * Sets the default formatter
+     * 
+     * @param instance {qx.log.appender.Formatter}
+     */
+    setDefaultFormatter: function(instance) {
```

Comment:
Yeah, this seems hokey to me. The default shouldn't be changeable; rather, there should be a default, I think, that `getFormatter` returns until `setFormatter` changes what is to be used.

Reply:
done

Reply:
No the test must be changed too.,.,.